include/exec/async_scope.hpp

```diff
@@ -543,38 +542,38 @@ namespace exec {
         }
 
         template <class _Tag, class... _As>
-        bool __save_completion(_Tag, _As&&... __as) noexcept {
```

Comment:
@kirkshoop do you know why`__save_completion` returns `bool` here? I can't figure out what it's used for, other than dropping the `__dispatch_result` call which leads to the waiter never being notified if emplacing into `__data_` raised an exception.